relayer/crates/starknet-madara-tests/src/impls/queries/tx_response.rs

```diff
@@ -49,3 +53,71 @@ where
         }
     }
 }
+
+// Madara returns a non-compliant JSON RPC response for the starknet_traceTransaction call.
+// Hence we need a workaround to parse the response manually here.
+// We may need further abstraction over the underlying JSON RPC client if there are other
+// non-compliant APIs out there.
+#[async_trait]
+pub trait CanTraceTransaction: HasAsyncErrorType {
+    async fn trace_transaction(&self, tx_hash: Felt) -> Result<TransactionTrace, Self::Error>;
+}
+
+impl<Chain> CanTraceTransaction for Chain
+where
+    Chain: HasRpcClient
+        + HasJsonRpcUrl
+        + CanRaiseAsyncError<serde_json::Error>
+        + CanRaiseAsyncError<reqwest::Error>,
+{
+    async fn trace_transaction(&self, tx_hash: Felt) -> Result<TransactionTrace, Self::Error> {
+        let params = TraceTransactionRequestRef {
+            transaction_hash: tx_hash.as_ref(),
+        };
+
+        let request_body = JsonRpcRequest {
+            id: 1,
+            jsonrpc: "2.0",
+            method: JsonRpcMethod::TraceTransaction,
+            params,
+        };
```

Comment:
can't we use https://docs.rs/starknet/0.13.0/starknet/providers/trait.Provider.html#tymethod.trace_transaction here?

Reply:
We can't, because the response returned by Madara does not follow the expected format, and result in "data did not match any variant of untagged enum JsonRpcResponse". See https://github.com/xJonathanLEI/starknet-rs/issues/514.

Reply:
even with `starknet_v13`? 😅 wow